pkg/fact/database/query.go

```diff
@@ -0,0 +1,61 @@
+package database
+
+import (
+	"errors"
+
+	"github.com/salsadigitalauorg/shipshape/pkg/connection"
+	"github.com/salsadigitalauorg/shipshape/pkg/fact"
+	"github.com/salsadigitalauorg/shipshape/pkg/utils"
+)
+
+type Query struct {
+	// Common fields.
+	Name           string          `yaml:"name"`
+	Format         fact.FactFormat `yaml:"format"`
+	ConnectionName string          `yaml:"connection"`
+	InputName      string          `yaml:"input"`
+	connection     connection.Connectioner
+	input          fact.Facter
+	errors         []error
+	data           interface{}
+
+	// Plugin fields.
+	Query string `yaml:"query"`
+}
+
+//go:generate go run ../../../cmd/gen.go fact-plugin --plugin=Query --package=database
+
+func init() {
+	fact.Registry["database.query"] = func(n string) fact.Facter { return &Query{Name: n} }
+}
+
+func (p *Query) PluginName() string {
+	return "database.query"
+}
+
+func (p *Query) SupportedConnections() (fact.SupportLevel, []string) {
+	return fact.SupportRequired, []string{"mysql"}
+}
+
+func (p *Query) SupportedInputs() (fact.SupportLevel, []string) {
+	return fact.SupportNone, []string{}
+}
+
+func (p *Query) Collect() {
+	mysqlConn := p.connection.(*connection.Mysql)
+	mysqlConn.Query = p.Query
```

Comment:
Does this guy escape the query strings? Can we pass direct user input to MySQL?

Reply:
This plugin is not available anymore - I created a db-search plugin instead, which does specifically that, using the goqu package.